Sources/S3Signer/S3Signer+Private.swift

```diff
@@ -72,8 +72,8 @@ extension S3Signer {
             let signHeaders = signed(headers: headers).encode(type: .queryAllowed) else {
                 throw Error.invalidEncoding
         }
-        let fullURL = "\(url.absoluteString)?x-amz-algorithm=AWS4-HMAC-SHA256&x-amz-credential=\(config.accessKey)%2F\(credScope)&x-amz-date=\(dates.long)&x-amz-expires=\(expiration.value)&x-amz-signedheaders=\(signHeaders)"
-        
+        let fullURL = "\(url.absoluteString)?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=\(config.accessKey)%2F\(credScope)&X-Amz-Date=\(dates.long)&X-Amz-Expires=\(expiration.value)&X-Amz-SignedHeaders=\(signHeaders)"
```

Comment:
Just a thought: maybe we should introduce an enum with all used headers. It should have a "case-intensity" functionality because some headers might be case insensitive.

Reply:
Yeah I thought about that as well. At the same time though, I'm not sure amazon will change the protocol for this anytime soon

Reply:
I think we should definitely enforce case sensitivity by either using an enum, throwing an error or sanitising the string if it breaks the functionality

Reply:
I mean if there is a rule on the input values, not the hardcoded stuff